builtin/providers/aws/resource_aws_lambda_permission.go

```diff
@@ -273,7 +276,8 @@ func resourceAwsLambdaPermissionDelete(d *schema.ResourceData, meta interface{})
 		policy := LambdaPolicy{}
 		err = json.Unmarshal(policyInBytes, &policy)
 		if err != nil {
-			return fmt.Errorf("Error unmarshalling Lambda policy: %s", err)
+			return resource.RetryableError(
+				fmt.Errorf("Error unmarshalling Lambda policy: %s", err))
```

Comment:
Here's a good example of something that probably should be NonRetryable but I left because I wanted this to be purely a refactor.